diff --git a/Cargo.toml b/Cargo.toml
index 400bbfeb..daae2397 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -79,7 +79,7 @@ predicates = "3"
 bluer = "0.17"
 hex = "0.4"
 async-trait = "0.1"
-libcosmic = { git = "https://github.com/Oppzippy/libcosmic.git" }
+libcosmic = { git = "https://github.com/Oppzippy/libcosmic.git", default-features = false }
 rusqlite = "0.38"
 syn = "2"
 quote = "1"
diff --git a/gui/Cargo.toml b/gui/Cargo.toml
index fb3a13e8..d786450c 100644
--- a/gui/Cargo.toml
+++ b/gui/Cargo.toml
@@ -18,7 +18,7 @@ tokio = { workspace = true, features = ["fs"] }
 futures = { workspace = true }
 libcosmic = { workspace = true, features = [
     "a11y",
-    "dbus-config",
+    "multi-window",
     "tokio",
     "winit",
     "wgpu",
